lib/rubygems/installer.rb

```diff
@@ -345,6 +345,8 @@ def install
 
     run_post_install_hooks
 
+    load_plugin
```

Comment:
If `run_post_install_hooks` happened before `load_plugin`, then any post install hooks potentially registered by the gem would be run immediately too, right?

For example, say the `rdoc` gem installs a plugin that generates documentation after gems are installed, then `gem install rdoc` would immediately generate docs for `rdoc` itself.

Did you consider this? Do you think it would be desirable?

Reply:
Ah, sorry. I didn't consider this.

`rdoc` (and `yard`) use `done_installing` hook not `post_install` hook but I can understand what you want to say.

I'll move `load_plugin` BEFORE `run_post_install_hooks`.

Reply:
Done.

BTW, I think that we can remove https://github.com/rubygems/rubygems/blob/master/lib/rubygems/rdoc.rb and and implement this by RubyGems plugin in RDoc after we merge this pull request.

Reply:
Oh, I didn't notice #6021.
I'll work on it after we merge this.

Reply:
Implementation: https://github.com/ruby/rdoc/pull/1171